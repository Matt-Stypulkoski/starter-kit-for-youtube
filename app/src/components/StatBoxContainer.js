import { React } from 'react';
import StatBox from './StatBox.js';

function StatBoxContainer(props) {
    return (
        <div className="stat-box-container">
            {props.statBoxList.map(function (statBox) {
                return <StatBox value={statBox[0]} label={statBox[1]} />
            })}
        </div>
    );
}

export default StatBoxContainer;