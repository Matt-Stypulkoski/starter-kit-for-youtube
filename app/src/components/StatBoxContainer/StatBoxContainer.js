import { React } from 'react';
import StatBox from '../StatBox/StatBox.js';

function StatBoxContainer(props) {
    return (
        <div className="stat-box-container">
            {props.statBoxList.map(function (statBox) {
                return <StatBox value={statBox.value} label={statBox.label} />
            })}
        </div>
    );
}

export default StatBoxContainer;