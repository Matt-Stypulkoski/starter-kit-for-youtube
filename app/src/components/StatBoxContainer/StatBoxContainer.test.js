import React from 'react';
import {cleanup, render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import StatBoxContainer from './StatBoxContainer';

afterEach(cleanup)

// jest.mock("./StatBox", () => () => <div data-testid="statbox"></div>);

it("renders", () => {
    const {statBoxContainer} = render(<StatBoxContainer statBoxList={[]} />);
});

it('displays props', () => {
    let mockStatBoxData = {
        "value": 20,
        "label": "test"
    }
    const tree = renderer.create(<StatBoxContainer statBoxList={[mockStatBoxData]} />).toJSON();
    expect(tree).toMatchSnapshot();
}); 
