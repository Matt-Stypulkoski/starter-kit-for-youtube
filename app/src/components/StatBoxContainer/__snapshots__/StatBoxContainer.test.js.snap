// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays props 1`] = `
<div
  className="stat-box-container"
>
  <div
    className="stat-box"
  >
    <h2
      className="stat-value"
    >
      20
    </h2>
    <h4
      className="stat-label"
    >
      test
    </h4>
  </div>
</div>
`;
